internal/ci/ci.go

```diff
@@ -1,6 +1,19 @@
 package ci
 
-import "os"
+import (
+	"fmt"
+	"os"
+	"reflect"
+)
+
+type CI struct {
+	Provider  Provider
+	OriginURL string
+	Repo      string
+	RefName   string // branch
```

Comment:
Not quite 💯  true, as github's and gitlab's refname is either a branch or a tag, depending on what's available:
```suggestion
	RefName   string // branch or tag
```